packages/context/src/inject.ts

```diff
@@ -129,6 +129,8 @@ export namespace inject {
     bindingKey: string,
     metadata?: Object,
   ) {
+    metadata = metadata || {};
+    metadata = Object.assign(metadata, {getter: true});
```

Comment:
FWIW, these two lines could be merged into a single one:

```ts
Object.assign({}, metadata, {getter:true});
```

(ditto for the next change below)